allennlp/data/instance.py

```diff
@@ -37,9 +38,13 @@ def index_fields(self, vocab: Vocabulary):
         """
         Converts all ``UnindexedFields`` in this ``Instance`` to ``IndexedFields``, given the
         ``Vocabulary``.  This `mutates` the current object, it does not return a new ``Instance``.
+        A ``DataIterator`` will call this on each pass through a dataset; we use the ``indexed``
```

Comment:
We need to be sure we'd only ever want  to index in-memory instances a single time. E.g what happens when we add to our test set vocab and we want to re-index - do we have to read the instances from disk again? Do we care about doing that?

Reply:
Ok chatting over lunch I don't think this is a problem anymore.